reframe/frontend/statistics.py

```diff
@@ -115,6 +117,43 @@ def failure_report(self):
         report.append(line_width * '-')
         return '\n'.join(report)
 
+    def failure_stats(self):
+        failures = {}
+        current_run = rt.runtime().current_run
+        for tf in (t for t in self.tasks(current_run) if t.failed):
+            check = tf.check
+            if tf.exc_info is not None:
+                from reframe.core.exceptions import format_exception
+            if tf.failed_stage not in failures:
+                failures[tf.failed_stage] = []
+            failures[tf.failed_stage].append(check.name)
+        line_width = 78
+        stats_start = line_width * '='
+        stats_title = 'FAILURE STATISTICS'
+        stats_end = line_width * '_'
+        stats_body = []
+        row_format = "{:<11} {:<5} {:<60}"
+        stats_hline = row_format.format(11*'-', 5*'-', 60*'-')
+        stats_header = row_format.format('Phase', '#', 'Failing tests')
+        total_num_tests = len(self.tasks(current_run))
+        total_num_failures = 0
+        for p in failures.keys():
+            total_num_failures += len(failures[p])
+        stats_body = ['']
+        stats_body.append('Total number of tests: %d' % int(total_num_tests))
+        stats_body.append('Total number of failures: %d' % 
+                           int(total_num_failures))
+        stats_body.append('')
+        stats_body.append(stats_header)
+        stats_body.append(stats_hline)
+        for p in failures.keys():
```

Comment:
Same here, iterate over the values directly:

```suggestion
        for f in failures.values():
```